services/QuillLMS/spec/models/teacher_activity_feed_spec.rb

```diff
@@ -27,7 +27,23 @@
       end
     end
 
-    context "activity_session's user gets after storage" do
+    context "activity_session without user" do
+      it 'shouldnt show in activity feed' do
+        TeacherActivityFeed.new(1).send(:delete_all)
+        activity_session_with_user = create(:activity_session, user: create(:user))
+        activity_session_without_user = create(:activity_session, user_id: nil)
```

Comment:
It looks like the activity_session factory has user assignment done indirectly via a [callback](url). A couple options are assign nil user after creation:

```
activity_session_without_user = create(:activity_session)
activity_session_without_user.update(user: nil)
```

or update the factory.

Reply:
Ah, yeah, somehow missed that. I'm going to delete this test, since I already have a test editing the session after the fact. I think that gives this feature enough coverage.